contracts/SpokePool.sol

```diff
@@ -282,11 +323,59 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     // Call this method to execute a leaf within the `distributionRoot` stored on this contract. Caller must include a
     // valid `inclusionProof` to verify that the leaf is contained within the root. The `relayerRefundId` is the index
     // of the specific distribution root containing the passed in leaf.
-    function distributeRelayerRefund(
+    // This method is internal because the implementing method must check the `distributionLeaf.amountToReturn` to see
+    // if this contract should send funds to L1, a process that is different for each L2.
+    function _distributeRelayerRefund(
         uint256 relayerRefundId,
         MerkleLib.DestinationDistribution memory distributionLeaf,
         bytes32[] memory inclusionProof
-    ) public {}
+    ) public {
+        require(distributionLeaf.chainId == chainId(), "Invalid chainId");
+
+        // Grab distribution root stored at `relayerRefundId`.
+        RelayerRefund storage refund = relayerRefunds[relayerRefundId];
+
+        // Check that `inclusionProof` proves that `distributionLeaf` is contained within the distribution root.
+        // Note: This should revert if the `distributionRoot` is uninitialized.
+        require(
+            MerkleLib.verifyRelayerDistribution(refund.distributionRoot, distributionLeaf, inclusionProof),
+            "Bad Proof"
+        );
+
+        // Verify the leafId in the leaf has not yet been claimed.
+        require(!MerkleLib.isClaimed(refund.claimedBitmap, distributionLeaf.leafId), "Already claimed");
+
+        // Set leaf as claimed in bitmap.
+        MerkleLib.setClaimed(refund.claimedBitmap, distributionLeaf.leafId);
+
+        // TODO: Should we be checking that distributionLeaf.refundAddresses.length == distributionLeaf.refundAmounts.length
+        // or can we assume that the `initiateRelayerRefund` on the HubPool would have been disputed on L1?
```

Comment:
As long as it's fairly cheap to check (which I think it is), I think this check makes sense to avoid any unexpected behavior below.

Reply:
Agreed